electron/helpers/utils.js

```diff
@@ -10,8 +11,45 @@ export {
   v4 as generateKey
 } from 'uuid'
 
+const harmfulSwitches = [
+  'remote-debugging-port',
+  'inspect',
+  'inspect-brk',
+  'inspect-brk-node',
+  'inspect-port',
+  'inspect-publish-uid',
+  'js-flags',
+  'proxy-server',
+  'host-rules'
+]
+
+function checkHarmfulSwitchPresence () {
+  for (harmfulSwitch of harmfulSwitches) {
+    if (
+      app.commandLine.hasSwitch(
+        harmfulSwitch
+      )
+    ) {
+      return true
+    }
+  }
+  return false
```

Comment:
@xyloflake I would suggest this:
```js
function isMatchedSwitch (
  switch
) {
  app
    .commandLine
    .hasSwitch(
      switch
    )
}

!!harmfulSwitches.find(
  isMatchedSwitch
)
```

Reply:
It's 18% slower at exiting the app, find method can't predict the outcome before. My method assumes it's true and preprocesses the entire thing so there's no memory freeze manipulation. Please refrain from using heavy methods in this function. Also, I'm not sure what 
```js
!!harmfulSwitches.find(
  isMatchedSwitch
)
```
does. Please clarify but do NOT use heavy methods like .find()

Reply:
I will wait with my review for this to be resolved

Reply:
@xyloflake https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find
Basically it stops when it finds the first match.

Reply:
I was talking about `!!` @staniel359

Reply:
@xyloflake It's turning a value to boolean.

Reply:
Got it, didn’t know that.

Reply:
I completely agree, as electron has much overhead itself, we should try to optimize the code as much as possible and reduce overhead.
I think you proposal was good, i dont see a need for expensive standard functions like find as it is not much better in terms of clean code and the `!!` looks a bit ridiculous but comes at a considerable performance cost.

Reply:
I completely agree with ‘!!’ expressing literally “not not” looking ridiculous XD

Also there are two intermediate conversion steps for that, again, 2 unnecessary steps are introduced. So performance is reduced.

Reply:
@xyloflake Do you know how often `.find()` is used in my code?
I don't think removind it from here would make much of a difference.
Please try to save megabytes, not kilobytes.

> Also the memory consumption of .find() is at least 3kb, vs my code would have 0.6kb memory usage, so the memory consumption is quite high

I won't comment on that.

Reply:
> Do you know how often .find() is used in my code?

No.

> I don't think removing it from here would make much of a difference.

It would. This is not on web (time and memory doesn’t matter there, at least not for you). This is the electron backend. This is before chromium loads. Even a single nanosecond counts here. It needs to quit ASAP. 

> Please try to save megabytes, not kilobytes.

Please elaborate.

Reply:
@xyloflake I use it quite often.

Look, as long as we're on Electron now, I think it wouldn't make any sense to improve this bloatware.
When/If we switch to Tauri, I'll give you the freedom to improve anything you want, okay?

Reply:
It's not about electron. It's about how quickly can we quit it? Please understand my concern, 1 clock cycle late and the fix won't have any value. I request you to not request this specific change, I'll comply on any other change not involving vulnerabilities, I promise. But changing this to the above code introduces a lot of risks, please try to understand that.

Reply:
@xyloflake Okay. Could you at least change it to `.forEach()`?

Reply:
@xyloflake Okay.

Reply:
@xyloflake One question though.
You're actually looping over all of the elements.
Could you at least break when you find the matched switch?

Reply:
@staniel359 for..of loop breaks automatically when it returns, that's one of the reasons I used for..of loop and not .forEach(), .forEach is a function which needs to be called explicitly and the inner variables have no point of returning beyond the scope of the forEach function. The stack of forEach is very limited which makes it incredibly slow, inefficient and inconsistent. 

One thing to note is for..of loop is newer than .forEach so it has better code in the v8 engine.

Also, .forEach is an Array/Object prototype while the for loop is a javascript inbuilt keyword, thus reducing ffi overheads.

For..of loop is 10 to 80 times faster than .forEach depending on how you implement it. And for loop automatically breaks when a return statement is used, so it's one instruction less than the .forEach prototype, thus being faster.

Reply:
@xyloflake Okay, sorry.
Somehow I thought that logic is the same.